### DISCUSSION TITLE
Chocolatey validation fails because of an invalid "packageSourceUrl"

### DISCUSSION THREAD

**Discussion by sualeh** [jreleaser/jreleaser#798] [Q&A] [answered]
@aalmiray - my push to Chocolatey failed validation because of an invalid "packageSourceUrl".

Please could you take a look at my [jreleaser.yml](https://github.com/schemacrawler/SchemaCrawler-Installers/blob/master/jreleaser.yml#L44-L50)? I am trying to have minimal configuration here. The generated "packageSourceUrl" does not exist.

```xml
<packageSourceUrl>https://github.com/schemacrawler/chocolatey-bucket</packageSourceUrl>
```

(@adriens)

**Comment by aalmiray** [comment, marked as answer]
The problem is found at the default template

https://github.com/jreleaser/jreleaser/blob/a2fed368653ffd6be8930eb6e5beec92a3cbf858/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/native-image/chocolatey/binary.nuspec.tpl#L20

It expects an URL to a bucket repository

https://github.com/jreleaser/jreleaser/blob/a2fed368653ffd6be8930eb6e5beec92a3cbf858/core/jreleaser-tools/src/main/java/org/jreleaser/packagers/ChocolateyPackagerProcessor.java#L107-L110

This works as long as the chocolatey build is remote, but your case is different as you don't have a bucket repository because you build and push on the same repository. This is a bug that has to be fixed however there are 2 options to workaround this problem:

1. generate a template that explicitly sets the value of the `<packageSourceUrl>` element.
2. set explicit values for `owner:` and `name:` in the chocolatey bucket configuration to match your repository.

Option 2 is the least intrusive.

**Comment by sualeh** [comment]
Thanks, @aalmiray - I wlll wait for a fix to #799

**Reply by aalmiray** [reply]
@sualeh the latest early-access build has the fix https://github.com/jreleaser/jreleaser/actions/runs/2017899702

**Reply by sualeh** [reply]
Thanks. I will try it out.

**Reply by aalmiray** [reply]
Looks like it's still building, sorry I was too fast and the build too slow.

**Reply by sualeh** [reply]
I already tried it out - https://community.chocolatey.org/packages/schemacrawler/16.16.12
If it fails Chocolatey validation, I will rerun

**Reply by aalmiray** [reply]
That is because the build has yet to finish. Please wait 30mins and try again

**Reply by sualeh** [reply]
Thanks, @aalmiray - it worked. cc:@adriens
https://community.chocolatey.org/packages/schemacrawler/16.16.12
